projects/dev/src/app/group/group-manage/pending-request/pending-request.component.ts

```diff
@@ -76,11 +149,73 @@ export class PendingRequestComponent implements OnInit, OnChanges {
   onExpandWidth(e) {}
 
   onClickAccept(e) {
-    this.onAccept.emit(e);
+    if (
+      this.selection.length === 0 ||
+      this.acceptLoading ||
+      this.rejectLoading
+    ) {
+      return;
+    }
+
+    this.acceptLoading = true;
+    this.groupService
+      .acceptJoinRequest(
+        this.id,
+        this.selection.map((val) => val.joining_user.group_id)
+      )
+      .subscribe(
+        (res) => {
+          this._manageRequestData(res, "accept", "accepted");
+          this.acceptLoading = false;
+        },
+        (err) => {
+          this._processRequestError(err, "accept");
+          this.acceptLoading = false;
+        }
+      );
   }
 
   onClickReject(e) {
-    this.onReject.emit(e);
+    if (
+      this.selection.length === 0 ||
+      this.acceptLoading ||
+      this.rejectLoading
+    ) {
+      return;
+    }
+
+    this.rejectLoading = true;
+    this.groupService
+      .rejectJoinRequest(
+        this.id,
+        this.selection.map((val) => val.joining_user.group_id)
+      )
+      .subscribe(
+        (res) => {
+          this._manageRequestData(res, "reject", "declined");
+          this.rejectLoading = false;
+        },
+        (err) => {
+          this._processRequestError(err, "reject");
+          this.rejectLoading = false;
+        }
+      );
+  }
```

Comment:
@smadbe 
You can review what I have done.
If you don't like it, we can discuss together.
I merged two handlers into one to avoid the duplicates.
Please let me know your ideas.